nav-app/src/app/viewmodels.service.ts

```diff
@@ -1652,8 +1652,11 @@ export class TechniqueVM {
         if (techniqueID !== undefined) this.techniqueID = techniqueID;
         else console.error("ERROR: TechniqueID field not present in technique")
 
-        if ("tactic" !== undefined) this.tactic = tactic;
-        else console.error("ERROR: tactic field not present in technique")
+        if (tactic !== undefined && tactic !== "") this.tactic = tactic;
+        else {
+            if ("comment" in obj && obj["comment"] !== undefined) alert(`WARNING: Tactic field for technique ID ${techniqueID} is empty, and will not be displayed in the matrix.`);
```

Comment:
Just a couple of suggestions for improvements to clarity and when the warning is displayed:
- The warning alert should be displayed any time the tactic is undefined or an empty string rather than being contingent on whether there is a comment on the technique, since there are other annotations that will be ignored as well
- I'd suggest modifying the warning text a bit- the the technique will still be displayed under the tactic and exist in the matrix. Only the technique annotations will be ignored, so I'd recommend something like:
```
WARNING: The tactic field on the technique <techniqueID> is not defined. Annotations for this technique will not be restored.
```